js/viz/chart.js

```diff
@@ -726,6 +727,119 @@ var dxChart = AdvancedChart.inherit({
+            let points = [];
+            const overloadedSeries = {};
+
+            series.forEach(s => {
+                const valAxis = s.getValueAxis();
+                const valVisualRange = valAxis.visualRange();
+                const valTranslator = valAxis.getTranslator();
+                const seriesIndex = that.series.indexOf(s);
+                const valViewPortFilter = rangeCalculator.getViewPortFilter(valVisualRange || {});
+
+                overloadedSeries[seriesIndex] = {};
+                series.forEach(sr => overloadedSeries[seriesIndex][that.series.indexOf(sr)] = 0);
+                const seriesPoints = [];
+
+                s.getPoints().filter(p => {
+                    return p.getOptions().visible && argViewPortFilter(p.argument) &&
+                        (valViewPortFilter(p.getMinValue(true)) || valViewPortFilter(p.getMaxValue(true)));
+                }).forEach(p => {
+                    const tp = {
+                        seriesIndex: seriesIndex,
+                        argument: p.argument,
+                        value: p.getMaxValue(true),
+                        size: p.bubbleSize || p.getOptions().size
+                    };
+                    if(p.getMinValue(true) !== p.getMaxValue(true)) {
+                        const mp = _extend({}, tp);
+                        mp.value = p.getMinValue(true);
+                        mp.x = argTranslator.to(mp.argument, 1);
+                        mp.y = valTranslator.to(mp.value, 1);
+                        seriesPoints.push(mp);
+                    }
+                    tp.x = argTranslator.to(tp.argument, 1);
+                    tp.y = valTranslator.to(tp.value, 1);
+                    seriesPoints.push(tp);
+                });
+
+                overloadedSeries[seriesIndex].pointsCount = seriesPoints.length;
+                overloadedSeries[seriesIndex].total = 0;
+                points = points.concat(seriesPoints);
+            });
+
+            const sortingCallback = argAxisType === "discrete" ?
+                (p1, p2) => argVisualRange.categories.indexOf(p1.argument) - argVisualRange.categories.indexOf(p2.argument) :
+                (p1, p2) => p1.argument - p2.argument;
+            points.sort(sortingCallback);
+
+            for(let i = 0; i < points.length; i++) {
+                const curPoint = points[i];
+                const size = curPoint.size;
+                const distance = i + 1 > points.length - i - 1 ? i + 1 : points.length - i;
+                if(_isDefined(curPoint.x) && _isDefined(curPoint.y)) {
+                    for(let j = 1; j < distance; j++) {
+                        const prevPoint = points[i - j];
+                        const nextPoint = points[i + j];
+                        const prev_x = _isDefined(prevPoint) ? prevPoint.x : null;
+                        const prev_y = _isDefined(prevPoint) ? prevPoint.y : null;
```

Comment:
const prev_x = _isDefined(prevPoint) ? prevPoint.x : null;
const prev_y = _isDefined(prevPoint) ? prevPoint.y : null;

it can be merged to one 'if'